rosidl_cmake/cmake/rosidl_generate_interfaces.cmake

```diff
@@ -125,22 +125,19 @@ macro(rosidl_generate_interfaces target)
     ament_index_register_resource("rosidl_interfaces" CONTENT "${_idl_files_lines}")
   endif()
 
-  # collect package names of recursive dependencies
+  # collect package names of recursive dependencies which contain interface files
   set(_recursive_dependencies)
   foreach(_dep ${_ARG_DEPENDENCIES})
-    list_append_unique(_recursive_dependencies "${_dep}")
+    if(DEFINED ${_dep}_INTERFACE_FILES)
+      list_append_unique(_recursive_dependencies "${_dep}")
+    endif()
     foreach(_dep2 ${${_dep}_RECURSIVE_DEPENDENCIES})
-      list_append_unique(_recursive_dependencies "${_dep2}")
+      if(DEFINED ${_dep2}_INTERFACE_FILES)
+        list_append_unique(_recursive_dependencies "${_dep2}")
+      endif()
     endforeach()
   endforeach()
 
-  # check that all dependencies are actually valid message packages
```

Comment:
This is not the case anymore. Now `rosidl_generator_runtime` exports a package which provides additional C++ header file(s) (containing the `BoundedVector`) necessary for the generated C++ messages to work.